### ISSUE TITLE
Bug with Inventory Tweaks Renewed mod

### ISSUE THREAD

**Issue by jahirtrap** [P3pp3rF1y/SophisticatedBackpacks#2]
middle click to sort the backpack inventory duplicates the updates, maybe the left interface is counted as an inventory space.

**Comment by P3pp3rF1y** [comment]
So this turns out to be an issue with inventory tweaks incorrectly updating slots that player cannot take anything from - in this case filter slots that are empty - if you fill the filter slots with items the issue doesn't happen. 

What actually happens is these get set to empty as part of invtweaks logic and that triggers the upgrade to save its state to the backpack and overrides the slot content that was already set to empty by inv tweaks logic before.

There could be a way to fix this on my side, but it would involve unnecessarily keeping track of old state of the slots so that I have something to understand if the new state is different and thus need to change which would allow to see that the old state is empty and new one is empty as well, but that's an addition of logic that's potentially error prone so I am not in favor of that.
It would be much better if inventory tweaks fixed the condition below so that it would honor the fact that slot can't be taken from and thus it's not a regular inventory slot and there's no sense in using it in the list of slots where inventory is to be sorted.

> (slot.canTakeStack(player) && slot.isItemValid(slot.getStack())) || !slot.getHasStack()

**Comment by P3pp3rF1y** [comment]
So I have spent like 3 hours today on trying to implemented something better that would prevent any mod from tempering with upgrade slots and the fake slots for filters, but that turned out to be a huge task that needs to duplicate a lot of functionality implemented in vanilla and then another task to implement interaction between the two different slot types as well. I have decided against it and have just added a small thing mentioned above that checks for previous state of upgrade filter slots and if both old and new is empty it doesn't save anything to the upgrade which prevents inventory tweaks from duping the upgrade.

But it doesn't prevent sorting of the upgrade slots into normal inventory so it's a bit annoying you still have to put upgrades back into upgrade slots after sorting the backpack. I wish inventory tweaks had compatibility api that it used which as far as I could see would make it possible to prevent this. 

Once I release new version I will close this issue.
